packages/astro/test/test-utils.js

```diff
@@ -141,13 +143,45 @@ export async function loadFixture(inlineConfig) {
 
 	let fixtureId = new Date().valueOf();
 	let devServer;
+	let checkServer;
 
 	return {
 		build: (opts = {}) => {
 			process.env.NODE_ENV = 'production';
 			return build(settings, { logging, telemetry, ...opts });
 		},
 		sync: (opts) => sync(settings, { logging, fs, ...opts }),
+		/**
+		 * This function runs the `check` command.
+		 * It injects the test dev server.
+		 *
+		 * @example
+		 * When used in a test suite:
+		 * ```js
+		 * 		let fixture = await loadFixture({
+		 * 			root: './fixtures/astro-check-watch/',
+		 * 		});
+		 * 		let { check, stop, watch } = fixture.check({
+		 * 			flags: { watch: true },
+		 * 		});
+		 * ```
+		 * @param opts {any} Override options sent to the check command
+		 */
+		check: async (opts) => {
+			checkServer = await check(settings, { logging, viteServer: devServer, ...opts });
+			return {
```

Comment:
any reason not to just return `checkServer` here? Since the object you are returning mirrors its API anyways.

Reply:
True, it's some leftover code. It makes sense to return the server itself! Good catch!